lib/constants.js

```diff
@@ -23,7 +23,7 @@ export const FILTER_DOMAIN_ONLY = 0;
 export const FILTER_COMBINED = 1;
 export const FILTER_RAW = 2;
 
-export const DOMAIN_REGEX = /((?=[a-z0-9-]{1,63}\.)(xn--)?[a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,63}/;
+export const DOMAIN_REGEX = /localhost:\d{1,5}|((?=[a-z0-9-]{1,63}\.)(xn--)?[a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,63}/;
```

Comment:
I'm not sure it matters, but perhaps moving the `localhost` match to the end would make it easier to read?  Dunno.

`export const DOMAIN_REGEX = /((?=[a-z0-9-]{1,63}\.)(xn--)?[a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,63}|localhost:\d{1,5}/;`

Reply:
I don't think the position really matters, but the port should be optional.

Reply:
Good point. 

This is updated: https://github.com/mikecao/umami/pull/173/files